diff --git a/arch/arm/dts/stm32mp157c-osd32mp1-brk-u-boot.dtsi b/arch/arm/dts/stm32mp157c-osd32mp1-brk-u-boot.dtsi
index 6154f02..99187ff 100644
--- a/arch/arm/dts/stm32mp157c-osd32mp1-brk-u-boot.dtsi
+++ b/arch/arm/dts/stm32mp157c-osd32mp1-brk-u-boot.dtsi
@@ -10,7 +10,6 @@
  */
 
 #include <dt-bindings/clock/stm32mp1-clksrc.h>
-#include "osd32mp1.dtsi"
 #include "stm32mp15-osd32mp1-ddr3-1x4Gb.dtsi"
 
 #include "stm32mp15-u-boot.dtsi"
@@ -27,7 +26,7 @@
        u-boot,boot-led = "LED2_GRN";
        u-boot,error-led = "LED2_RED";
        u-boot,mmc-env-partition = "ssbl";
-       st,stm32prog-gpios = <&gpiod 9 GPIO_ACTIVE_LOW>;
+       st,stm32prog-gpios = <&gpiod 9 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
    };
 
    clocks {
